src/group_theory/perm/cycles.lean

```diff
@@ -66,6 +66,32 @@ by classical; exact ⟨(n % order_of f).to_nat, by {
   have := n.mod_nonneg (int.coe_nat_ne_zero.mpr (ne_of_gt (order_of_pos f))),
   rwa [← gpow_coe_nat, int.to_nat_of_nonneg this, ← gpow_eq_mod_order_of] }⟩
 
+/-- The subgroup generated by a cycle is in bijection with its support -/
+noncomputable def is_cycle.gpowers_equiv_support {σ : perm α} (hσ : is_cycle σ) :
+  (↑(subgroup.gpowers σ) : set (perm α)) ≃ (↑(σ.support) : set α) :=
+equiv.of_bijective (λ τ, ⟨τ (classical.some hσ), by
+{ obtain ⟨τ, n, rfl⟩ := τ,
+  rw [finset.mem_coe, mem_support],
+  exact λ h, (classical.some_spec hσ).1
+    ((σ ^ n).injective (by rwa [←mul_apply, mul_gpow_self, ←mul_self_gpow])) }⟩) ⟨by
+{ rintros ⟨a, m, rfl⟩ ⟨b, n, rfl⟩ h,
+  ext y,
+  by_cases hy : σ y = y,
+  { simp_rw [subtype.coe_mk, gpow_apply_eq_self_of_apply_eq_self hy] },
+  { obtain ⟨i, rfl⟩ := (classical.some_spec hσ).2 y hy,
+    rw [subtype.coe_mk, subtype.coe_mk, gpow_apply_comm σ m i, gpow_apply_comm σ n i],
+    exact congr_arg _ (subtype.ext_iff.mp h) } }, by
+{ rintros ⟨y, hy⟩,
+  rw [finset.mem_coe, mem_support] at hy,
+  obtain ⟨n, rfl⟩ := (classical.some_spec hσ).2 y hy,
+  exact ⟨⟨σ ^ n, n, rfl⟩, rfl⟩ }⟩
```

Comment:
I'm not entirely sure the following is the best way to write the proofs, but I like it better than having the `by` on the "previous" lines.
```suggestion
equiv.of_bijective
  (λ τ, ⟨τ (classical.some hσ),
    by { obtain ⟨τ, n, rfl⟩ := τ,
         rw [finset.mem_coe, mem_support],
         exact λ h, (classical.some_spec hσ).1
           ((σ ^ n).injective (by rwa [←mul_apply, mul_gpow_self, ←mul_self_gpow])) }⟩)
  begin
    split,
    { rintros ⟨a, m, rfl⟩ ⟨b, n, rfl⟩ h,
      ext y,
      by_cases hy : σ y = y,
      { simp_rw [subtype.coe_mk, gpow_apply_eq_self_of_apply_eq_self hy] },
      { obtain ⟨i, rfl⟩ := (classical.some_spec hσ).2 y hy,
        rw [subtype.coe_mk, subtype.coe_mk, gpow_apply_comm σ m i, gpow_apply_comm σ n i],
        exact congr_arg _ (subtype.ext_iff.mp h) } },
    { rintros ⟨y, hy⟩,
      rw [finset.mem_coe, mem_support] at hy,
      obtain ⟨n, rfl⟩ := (classical.some_spec hσ).2 y hy,
      exact ⟨⟨σ ^ n, n, rfl⟩, rfl⟩ }
  end
```